allensdk/brain_observatory/behavior/behavior_project_cache/behavior_ecephys_project_cache.py

```diff
@@ -0,0 +1,283 @@
+from functools import partial
+from typing import Optional, Union
+from pathlib import Path
+import logging
+
+from allensdk.brain_observatory.behavior.behavior_project_cache.project_apis.data_io import (  # noqa: E501
+    BehaviorEcephysProjectCloudApi)
+from allensdk.api.warehouse_cache.caching_utilities import call_caching
+
+
+class VisualBehaviorEcephysProjectCache(object):
+
+    def __init__(
+            self,
+            fetch_api: Optional[BehaviorEcephysProjectCloudApi] = None,
+            fetch_tries: int = 2,
+            ):
+        """ Entrypoint for accessing Visual Behavior Neuropixels data.
+
+        Supports access to metadata tables:
+        get_ecephys_session_table()
+        get_behavior_session_table()
+        get_probe_table()
+        get_channel_table()
+        get_unit_table
+
+        Provides methods for instantiating session objects
+        from the nwb files:
+        get_ecephys_session() to load BehaviorEcephysSession
+        get_behavior_sesion() to load BehaviorSession
+
+        Provides tools for downloading data:
+
+        Will download data from the s3 bucket if session nwb file is not
+        in the local cache, othwerwise will use file from the cache.
+
+        Parameters
+        ==========
+        fetch_api :
+            Used to pull data from remote sources, after which it is locally
+            cached.
+        fetch_tries :
+            Maximum number of times to attempt a download before giving up and
+            raising an exception. Note that this is total tries, not retries.
+            Default=2.
+        """
+
+        self.fetch_api = fetch_api
+        self.cache = None
+
+        self.fetch_tries = fetch_tries
+        self.logger = logging.getLogger(self.__class__.__name__)
+
+    @property
+    def manifest(self):
+        if self.cache is None:
+            api_name = type(self.fetch_api).__name__
+            raise NotImplementedError(f"A {type(self).__name__} "
+                                      f"based on {api_name} "
+                                      "does not have an accessible manifest "
+                                      "property")
+        return self.cache.manifest
+
+    @classmethod
+    def from_s3_cache(cls, cache_dir: Union[str, Path],
+                      bucket_name: str = "visual-behavior-ecephys-data",
```

Comment:
If, as I claim, this class really should only be supporting the VBN data release, then there's probably no need to expose bucket_name and project_name to the users. Those should be hard-coded into the class. Future cache classes which are meant to support future releases would hard-code in different values of bucket_name and project_name.

I'm open to pushback on this.

Reply:
I made them class variables.